### DISCUSSION TITLE
possible solution for faster local updates

### DISCUSSION THREAD

**Discussion by DietmarHoch** [fabian-lauer/dbus-solax-x1-pvinverter#2] [Ideas]
Hi,

great Job so far.
If you seach for an solution for faster updates without solax cloud,
you should take a look at this:
https://github.com/simatec/ioBroker.solax
Installed it today, but without sun, not possible to test it.
Dont forget to downgrade the Wifi Dongle.

Greetings from Jena
Dietmar

**Comment by DietmarHoch** [comment]
Hi again,

i was testing it with:
**curl -X POST http://192.168.178.43/?optType=ReadRealTimeData'&'pwd=admin --data "string" -H "Accept: application/json, txt/plain, */*\r\n" -H "X-Forwarded-For: 5.8.8.8\r\n"**

ip and pw should be changed to your system

**Reply by fabian-lauer** [reply]
Hi Dietmar - will test in later - thanks for sharing 👍

**Reply by DietmarHoch** [reply]
Hi,

please try this

headers = {                                                                                                                              
'Accept': 'application/json', 'X-Forwarded-For': '5.8.8.8',                                                                              
}                                                                                                                                        
params = {                                                                                                                               
    'optType': 'ReadRealTimeData',                                                                                                       
    'pwd': 'admin',                                                                                                                      
}                                                                                                                                        
                                                                                                                                         
data_r = requests.post('http://192.168.178.43/', headers=headers, params=params)                                                         
                                                                                                                                         
the answer is like this:
{"type":"X1-Boost-Air-Mini","SN":"XYZ7K9TKZZ","ver":"2.033.20","Data":[0.1,0.0,181.8,0.0,0.4,233.5,13,18,3.0,103.1,-166,23,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1.50,1716.72,0,0,0,0,0,0,0,49.98,0,0,0,0,0,0,0,0,0,0.00,0,0,0,0,0.00,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"Information":[1.500,4,"X1-Boost-Air-Mini","XMT152F6215111",1,2.19,0.00,1.37,0.00]}

**Comment by DietmarHoch** [comment]
Hi,

these Values from the json work if data comes local

          self._lastLocalUpdate = time.time()
          self._lastLovalACVoltage = meter_data['Data'][5]
          self._lastLocalACCurrent = meter_data['Data'][4]
          self._lastLocalACPower = meter_data['Data'][6]
          self._lastLocalACEnergyTotal = meter_data['Data'][9]
          self._lastLocalInverterStatus = int(meter_data['Data'][68])

problem is at night when inverter is offline and  will give thir error:
HTTPConnectionPool(host='192.168.178.43', port=80): Max retries exceeded with url: /?optType=ReadRealTimeData&pwd=admin (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7592b730>: Failed to establish a new connection: [Errno 113] No route to host'))

**Reply by fabian-lauer** [reply]
Hi,

wow great work, also for mapping the values.
Regarding the exception - you could catch it and handle differently:
- maybe quit the script - the deamon will restart it
- or just wait a bit longer (5 Min) until next retry
- or simply keep it as it is - and set values to 0
